### PR TITLE
gamm keeper delete obsolete files

### PR DESCRIPTION
**Pull request by AlpinYukseloglu**
Closes: #2155

## What is the purpose of the change

Cleanup of `x/gamm/keeper`

## Brief Changelog

- [x] Lets view keeper.go as "generic SDK boilerplate", and move params.go code into keeper.go.
- [x] Remove usages of poolNumber everywhere, have it only be poolId.
- [x] Move SetStableSwapScalingFactors to stableswap
- [x]  Lets delete marshal_bench_test.go - this existed for some misc. historical reason
- [x]  Delete grpc_query_internal_test.go

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

**Review by mattverse — APPROVED** [review]
Changelog entry should be changed to breaking changes, other than that LGTM!

**Review by ValarDragon — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -66,7 +66,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * [#1630](https://github.com/osmosis-labs/osmosis/pull/1630) Delete the v043_temp module, now that we're on an updated SDK version.
 * [#1667](https://github.com/osmosis-labs/osmosis/pull/1673) Move wasm-bindings code out of app package into its own root level package.
 * [#2013](https://github.com/osmosis-labs/osmosis/pull/2013) Make `SetParams`, `SetPool`, `SetTotalLiquidity`, and `SetDenomLiquidity` GAMM APIs private
-*[#1857](https://github.com/osmosis-labs/osmosis/pull/1857) x/mint rename GetLastHalvenEpochNum to GetLastReductionEpochNum
+* [#1857](https://github.com/osmosis-labs/osmosis/pull/1857) x/mint rename GetLastHalvenEpochNum to GetLastReductionEpochNum
+* [#2160](https://github.com/osmosis-labs/osmosis/pull/2160) Clean up GAMM keeper (move `x/gamm/keeper/params.go` contents into `x/gamm/keeper/keeper.go`, replace all uses of `PoolNumber` with `PoolId`, move `SetStableSwapScalingFactors` to stableswap package, and delete marshal_bench_test.go and grpc_query_internal_test.go)
```
[the remark this answers is not in the document]
Comment (mattverse) [reply CHANGELOG.md:70]: @ValarDragon but the keys that changed are used directly in state though aren't they?

[the remark this answers is not in the document]
Comment (ValarDragon) [reply CHANGELOG.md:70]: No key changed? Only the variable name for it changed

[the remark this answers is not in the document]
Comment (ValarDragon) [reply CHANGELOG.md:70]: https://github.com/osmosis-labs/osmosis/pull/2160/files#diff-c5fa46e1fb34134cb3ff99eedd1466cf5eefb742d1e28c4760abd946479f1953R23

[the remark this answers is not in the document]
Comment (ValarDragon) [reply CHANGELOG.md:70]: It does break the genesis json serialization, which would just affect testnets.

File: proto/osmosis/gamm/v1beta1/genesis.proto [hunk 2]
```diff
@@ -21,6 +21,6 @@ option go_package = "github.com/osmosis-labs/osmosis/v10/x/gamm/types";
 message GenesisState {
   repeated google.protobuf.Any pools = 1
       [ (cosmos_proto.accepts_interface) = "PoolI" ];
-  uint64 next_pool_number = 2;
+  uint64 next_pool_id = 2;
```
Comment (alexanderbez) [inline proto/osmosis/gamm/v1beta1/genesis.proto:24]: AFAIU, when you break proto like this, you technically have to bump the package version.

Comment (ValarDragon) [reply proto/osmosis/gamm/v1beta1/genesis.proto:24]: So field name changes like this break clients and json serialization, but I think its fine to do so for genesis

cc @daniel-farina  @czarcas7ic if field renames / genesis file breaks like this end up being problematic for integrators

**Comment by ValarDragon** [comment]
Can we separate out the deletes here from the genesis field rename

Can easily get all the file deletes merged

File: proto/osmosis/gamm/v1beta1/genesis.proto (hunk 2 above)
Comment (ValarDragon) [reply proto/osmosis/gamm/v1beta1/genesis.proto:24]: Actually I'll just add a json serialization name override, so this then won't break anything

**Comment by ValarDragon** [comment]
Actually I'll just commit undoing that proto field rename, and we can re-open a new issue for talking about genesis breaks

**Pull request merged**